docs/design.md

```diff
@@ -0,0 +1,110 @@
+# Design and architecture
+
+## Authors
+ * [Arron Wang](https://github.com/arronwy)
+ * [Jiang Liu](https://github.com/jiangliu)
+
+## Design
+The `image-rs` crate is a rustified and tailored version of [containers/image](https://github.com/containers/image),
+to provide a small, simple, secure, lightweight and high performance OCI container image management library for
+[Confidential Containers](https://github.com/confidential-containers).
+
```

Comment:
I know this is the main focus of this crate at the moment, but we should make that statement less strong. Something like:

```
...a small, simple, secure, lightweight and high performance OCI container image management library. The crate development is driven by the Confidential Containers project scope but it might be used in a more generic fashion.
```

Reply:
Yes, make sense, I'll update this part.